src/recipe/recipe.controller.ts

```diff
@@ -58,37 +58,29 @@ export class RecipeController {
     return await this.recipeService.updateRecipe(Number(recipeId), newRecipe);
   }
 
-  @Get('/id/:id')
-  @ApiOperation({ summary: 'Get one recipe by id' })
-  @ApiResponse({ type: RecipeResponse })
-  @ApiParam({ name: 'id', required: true })
-  async getOneRecipe(@Param('id') recipeId: number): Promise<RecipeResponse> {
-    return await this.recipeService.findRecipeById(Number(recipeId));
-  }
-
-  @Get('/list')
-  @UseGuards(AuthGuard('jwt'), IsAdminGuard)
+  @Get()
+  @UseGuards(AuthGuard('jwt'))
   @ApiOperation({ summary: 'Get all recipes' })
   @ApiResponse({ type: [RecipeResponse] })
   @ApiUnauthorizedResponse({ description: 'Not logged in' })
   @ApiForbiddenResponse({
     description: 'User is not an admin',
   })
   async getAllRecipes(
-    @Query('page') page: string,
-    @Query('limit') limit: string,
+    @Query() query: ListRecipesQuery,
```

Comment:
`ListRecipesQuery` that you have used in an interface. 
You should use class that contains properties, validation and swagger decorators.
you can refer to your existing classes that define DT's (eg CreateRecipeBody) or:

https://stackoverflow.com/questions/70225539/how-validate-query-params-in-nestjs
https://docs.nestjs.com/techniques/validation